images/linux/scripts/installers/codeql-bundle.sh

```diff
@@ -4,22 +4,37 @@
 ##  Desc:  Install the CodeQL CLI Bundle to the toolcache.
 ################################################################################
 
-source $HELPER_SCRIPTS/install.sh
+source "$HELPER_SCRIPTS/install.sh"
 
 # Retrieve the name of the CodeQL bundle preferred by the Action (in the format codeql-bundle-YYYYMMDD).
-codeql_bundle_name="$(curl -sSL https://raw.githubusercontent.com/github/codeql-action/v2/src/defaults.json | jq -r .bundleVersion)"
-# Convert the bundle name to a version number (0.0.0-YYYYMMDD).
-codeql_bundle_version="0.0.0-${codeql_bundle_name##*-}"
+defaults="$(curl -sSL https://raw.githubusercontent.com/github/codeql-action/v2/src/defaults.json)"
+codeql_bundle_name="$(${defaults} | jq -r .bundleVersion)"
+codeql_cli_version="$(${defaults} | jq -r .cliVersion)"
+prior_codeql_bundle_name="$(${defaults} | jq -r .priorBundleVersion)"
+prior_codeql_cli_version="$(${defaults} | jq -r .priorCliVersion)"
 
-extraction_directory="$AGENT_TOOLSDIRECTORY/CodeQL/$codeql_bundle_version/x64"
-mkdir -p "$extraction_directory"
+# Convert the bundle names to bundles with a version number (x.y.z-YYYYMMDD).
+codeql_bundle_version="${codeql_cli_version}-${codeql_bundle_name##*-}"
+prior_codeql_bundle_version="${prior_codeql_cli_version}-${prior_codeql_bundle_name##*-}"
 
-echo "Downloading CodeQL bundle $codeql_bundle_version..."
-download_with_retries "https://github.com/github/codeql-action/releases/download/$codeql_bundle_name/codeql-bundle.tar.gz" "/tmp" "codeql-bundle.tar.gz"
-tar -xzf "/tmp/codeql-bundle.tar.gz" -C "$extraction_directory"
+# For both of the bundle versions, download and name appropriately.
+for bundle in "${codeql_bundle_version} ${codeql_bundle_name}" "${prior_codeql_bundle_version} ${prior_codeql_bundle_name}"
+do
+    set -- "$bundle"
 
-# Touch a special file that indicates to the CodeQL Action that this bundle was baked-in to the hosted runner images.
-touch "$extraction_directory/pinned-version"
+    echo "Downloading CodeQL bundle $1..."
+    download_with_retries "https://github.com/github/codeql-action/releases/download/$2/codeql-bundle.tar.gz" "/tmp" "codeql-bundle.tar.gz"
+    codeqlArchive="/tmp/codeql-bundle.tar.gz"
 
-# Touch a file to indicate to the toolcache that setting up CodeQL is complete.
-touch "$extraction_directory.complete"
+    codeqlToolcachePath="$AGENT_TOOLSDIRECTORY/codeql/$1/x64"
+    mkdir -p "$codeqlToolcachePath"
+
+    echo "Unpacking the downloaded CodeQL bundle archive..."
+    tar -xzf "$codeqlArchive" -C "$codeqlToolcachePath"
+
+    # Touch a special file that indicates to the CodeQL Action that this bundle was baked-in to the hosted runner images.
+    touch "$codeqlToolcachePath/pinned-version"
```

Comment:
I believe this only affects a codepath that relies on the toolcache only containing a single version of `CodeQL`.  Therefore we should be able to remove this. https://github.com/github/codeql-action/blob/cf1437a51414bcd7b1ce9aa9704fa8b55b2c2926/src/codeql.ts#L477

Reply:
Oh, interesting, nice catch!

Reply:
Done!

Reply:
On reflection, I think we want to keep this, but only for the latest CodeQL version.  This means we can continue supporting the behaviour on Enterprise where the toolcache version of CodeQL can override the defaults.json version on Enterprise.  We'll just need to modify the CodeQL Action to allow the override when the toolcache contains a single pinned version of CodeQL, vs just a single version of CodeQL.

Reply:
Oh, I see, that makes sense. Thanks for explaining the logic! I'll make the change 👍